src/popup/components/LatestTransactionsCard.vue

```diff
@@ -113,21 +112,28 @@ export default defineComponent({
       ))
         .sort(defaultTransactionSortingCallback)
         .slice(0, DASHBOARD_TRANSACTION_LIMIT);
-    }
-
-    // This replaces it poling
-    watch(balances, updateData);
 
-    onMounted(async () => {
-      await updateData();
       isLoading.value = false;
-    });
+    }
+
+    // To avoid unnecessary data transfers instead of constant polling
+    // we are fetching the transactions only if the total balance of the accounts changes.
+    watch(
+      balancesTotal,
+      (val, oldVal) => {
+        if (!latestTransactions.value.length || val !== oldVal) {
+          setTimeout(() => updateData(), 100);
```

Comment:
Transaction appearing on the middleware slower than node updates the balance. So the latest transaction will not be fetched in most of cases.

Reply:
As discussed on priv I increased the delay. In the future we need to replace this with websockets.